### PR TITLE
pkg/install: delete target dir upon directory move

### LINKED ISSUE #55: plugin install is broken in master: rename fails "file exists"
I added `bin: ca-cert.sh` to [existing ca-cert.yaml](https://github.com/GoogleContainerTools/krew-index/blob/4bcf63c95a3f42a37fbaca1d735f2efb981d4d65/plugins/ca-cert.yaml) and ran:

```
kubectl krew install --source ~/.krew/index/plugins/ca-cert.yaml -v=10

I0925 10:43:18.446458   77773 git.go:62] Going to run git pull --ff-only -v
From https://github.com/GoogleContainerTools/krew-index
 = [up to date]      master     -> origin/master
Already up to date.
Updated index
Will install plugin: ca-cert
I0925 10:43:18.943571   77773 install.go:105] Installing plugin: ca-cert
I0925 10:43:18.943610   77773 install.go:69] Looking for installed versions
I0925 10:43:18.943639   77773 util.go:61] Searching for installed versions of ca-cert in "/Users/ahmetb/.krew/bin"
I0925 10:43:18.943675   77773 install.go:78] Finding download target for plugin ca-cert
I0925 10:43:18.943689   77773 util.go:43] Matching platform for labels(arch=amd64,os=darwin)
I0925 10:43:18.943792   77773 util.go:50] Found matching platform with index (0)
I0925 10:43:18.943817   77773 util.go:113] Matching plugin version is 8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c
I0925 10:43:18.943827   77773 install.go:46] Creating download dir "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew/ca-cert"
I0925 10:43:18.944054   77773 install.go:56] Getting sha256 (8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c) signed version
I0925 10:43:18.944087   77773 downloader.go:32] Fetching "https://github.com/ahmetb/kubectl-extras/archive/c403c5714a4442b8314e5a86b0ff63f7a815f83f.zip"
I0925 10:43:19.750567   77773 downloader.go:39] Reading download data into memory
I0925 10:43:19.818536   77773 downloader.go:44] Read 19613 bytes of download data into memory
I0925 10:43:19.818588   77773 downloader.go:52] Extracting download zip to "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew/ca-cert"
I0925 10:43:19.829590   77773 move.go:155] Creating plugin dir "/Users/ahmetb/.krew/store/ca-cert"
I0925 10:43:19.829832   77773 move.go:161] Creating temp plugin move operations dir "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move184368352"
I0925 10:43:19.829850   77773 move.go:126] Finding move targets from "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew/ca-cert" to "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move184368352" with file operation=index.FileOperation{From:"/*/ca-cert/*", To:"."}
I0925 10:43:19.829898   77773 move.go:44] Trying to move single file directly from="/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew/ca-cert" to="/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move184368352" with file operation=index.FileOperation{From:"/*/ca-cert/*", To:"."}
I0925 10:43:19.830031   77773 move.go:52] Wasn't a single file, proceeding with Glob move
I0925 10:43:19.830323   77773 move.go:76] Move operations are complete
I0925 10:43:19.830338   77773 move.go:133] Move file from "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew/ca-cert/kubectl-extras-c403c5714a4442b8314e5a86b0ff63f7a815f83f/ca-cert/ca-cert.bash" to "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move184368352/ca-cert.bash"
I0925 10:43:19.830543   77773 move.go:133] Move file from "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew/ca-cert/kubectl-extras-c403c5714a4442b8314e5a86b0ff63f7a815f83f/ca-cert/plugin.yaml" to "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move184368352/plugin.yaml"
I0925 10:43:19.830729   77773 move.go:172] Move "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move184368352" to "/Users/ahmetb/.krew/store/ca-cert/8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c"
W0925 10:43:19.836107   77773 install.go:112] failed to install plugin "ca-cert", err: failed to dowload and move during installation, err: could not rename file from "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move184368352" to "/Users/ahmetb/.krew/store/ca-cert/8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c", err: rename /var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move184368352 /Users/ahmetb/.krew/store/ca-cert/8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c: file exists
F0925 10:43:19.836154   77773 root.go:48] failed to install some plugins: [ca-cert]
```

### PR DESCRIPTION
**Pull request by ahmetb**
fixes #55

### CODE REVIEW

**Comment by ahmetb** [comment]
now instead of failing like in #55, this proceeds:

```
I0925 10:58:15.889510   79856 move.go:172] Move directory "/var/folders/76/5g_jbgjn6xg_jkwcm2n_m52r00dl55/T/krew-temp-move716852439" to "/Users/ahmetb/.krew/store/ca-cert/8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c"
I0925 10:58:15.889566   79856 move.go:190] there's already a directory at move target "/Users/ahmetb/.krew/store/ca-cert/8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c". deleting.
I0925 10:58:15.890531   79856 move.go:194] move target directory "/Users/ahmetb/.krew/store/ca-cert/8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c" cleaned up
I0925 10:58:15.896237   79856 install.go:137] Creating symlink from "/Users/ahmetb/.krew/store/ca-cert/8be8ed348d02285abc46bbf7a4cc83da0ee9d54dc2c5bf86a7b64947811b843c/ca-cert.bash" to "/Users/ahmetb/.krew/bin/kubectl-ca-cert"
I0925 10:58:15.896388   79856 install.go:141] Created symlink in "/Users/ahmetb/.krew/bin/kubectl-ca-cert"
Installed plugin: ca-cert
```

However, I'm not sure if I just added this redundant code: Somehow there were already files in `store/ca-cert` dir but the installation/symlinking previously failed I guess. 

Adding this `os.RemoveAll(to)` helped it.

File: pkg/installation/move.go [hunk 1]
```diff
@@ -169,19 +169,32 @@ func moveToInstallDir(download, pluginDir, version string, fos []index.FileOpera
...
-// moveOrCopy will try to rename a dir or file. If rename is not supported a manual copy will be performed.
-func moveOrCopy(from, to string) error {
+// moveOrCopyDir will try to rename a dir or file. If rename is not supported a
+// manual copy will be performed. Existing files at "to" will be deleted
+func moveOrCopyDir(from, to string) error {
 	// Try atomic rename (does not work cross partition).
-	err := os.Rename(from, to)
+	fi, err := os.Stat(to)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("error checking move target dir %q: %+v", to, err)
```
Comment (lbb) [inline pkg/installation/move.go]: , err: %+v” also for other fmt calls

File: pkg/installation/move.go [hunk 2]
```diff
@@ -169,19 +169,32 @@ func moveToInstallDir(download, pluginDir, version string, fos []index.FileOpera
...
+	}
+	if fi != nil && fi.IsDir() {
+		glog.V(4).Infof("there's already a directory at move target %q. deleting.", to)
+		if err := os.RemoveAll(to); err != nil {
+			return fmt.Errorf("error cleaning up dir %q: %+v", to, err)
+		}
+		glog.V(4).Infof("move target directory %q cleaned up", to)
```
Comment (lbb) [inline pkg/installation/move.go]: Capitalize, also other golg calls

File: pkg/installation/move.go (hunk 1 above)
Comment (ahmetb) [reply pkg/installation/move.go]: I don't think the `err:` part is necessary.
stdlib or most 3rd party packages doesn't use this format. it's more like:

    something bad happened: [REASON]

**Pull request merged**